test/unit/connection/ClientMessageReaderTestRenewed.js

```diff
@@ -0,0 +1,155 @@
+
+        const buffer = writeToBuffer(message);
+        reader.append(buffer);
+
+        const messageRead = reader.read();
+        expect(messageRead.hasNextFrame()).to.be.true;
+        let frameRead = messageRead.nextFrame();
+        expect(frameRead.content).to.deep.equal(frame1.content);
+
+        expect(messageRead.hasNextFrame()).to.be.true;
+        frameRead = messageRead.nextFrame();
+        expect(frameRead.content).to.deep.equal(frame2.content);
+
+        expect(messageRead.hasNextFrame()).to.be.true;
+        frameRead = messageRead.nextFrame();
+        expect(frameRead.content).to.deep.equal(frame3.content);
+
+        expect(messageRead.hasNextFrame()).to.be.false;
+    });
+
+    it('reads frames in multiple calls', function() {
+        const frame = createFrameWithRandomBytes(1000);
+
+        const message = ClientMessage.createForEncode(0, 0, 0);
+        message.addFrame(frame);
+
+        const buffer = writeToBuffer(message);
+
+        const firstPartition = buffer.slice(0, 750);
+        const secondPartition = buffer.slice(750);
+
+        reader.append(firstPartition);
+        reader.append(secondPartition);
+
+        const messageRead = reader.read();
+        expect(messageRead.hasNextFrame()).to.be.true;
+
+        const frameRead = messageRead.nextFrame();
+        expect(frameRead.content).to.deep.equal(frame.content);
+
+        expect(messageRead.hasNextFrame()).to.be.false;
+    });
+
+    it('reads frames in multiple calls when last piece is small', function() {
+        const frame = createFrameWithRandomBytes(1000);
+
+        const message = ClientMessage.createForEncode(0, 0, 0);
+        message.addFrame(frame);
+
+        const buffer = writeToBuffer(message);
+
+        const firstPartition = buffer.slice(0, 750);
+        const secondPartition = buffer.slice(750, 1002);
+        const thirdPartition = buffer.slice(1002);
+
+        reader.append(firstPartition);
+        reader.append(secondPartition);
+        reader.append(thirdPartition);
+
+        const messageRead = reader.read();
+        expect(messageRead.hasNextFrame()).to.be.true;
+
+        const frameRead = messageRead.nextFrame();
+        expect(frameRead.content).to.deep.equal(frame.content);
+
+        expect(messageRead.hasNextFrame()).to.be.false;
+    });
+
+    it('reads when the Frame length and Flags not received at first', function() {
+        const frame = createFrameWithRandomBytes(100);
+
+        const message = ClientMessage.createForEncode(0, 0, 0);
+        message.addFrame(frame);
+
+        const buffer = writeToBuffer(message);
+        const capacity = buffer.length;
+        let limit = 4;
+
+        const firstPartition = buffer.slice(0, limit);
+
+        reader.append(firstPartition);
+
+        let messageRead = reader.read();
+        expect(messageRead).to.be.null;
+
+        limit = capacity;
+        const secondPartition = buffer.slice(4, limit);
+        reader.append(secondPartition);
```

Comment:
```suggestion
        const buffer = writeToBuffer(message);
        const firstPartition = buffer.slice(0, 4);
        reader.append(firstPartition);

        let messageRead = reader.read();
        expect(messageRead).to.be.null;

        const secondPartition = buffer.slice(4, buffer.length);
        reader.append(secondPartition);
```

also note that `capacity != length`. Node.js buffers does not have capacity as I know because they are not resizable. Length will work fine here though.

Reply:
I see. I was not sure while doing this. Thanks! Done.